Dockerfile

```diff
@@ -1,24 +1,32 @@
-FROM puckel/docker-airflow:1.10.9
-USER root
+FROM apache/airflow:2.1.2-python3.8
 ENV POETRY_VIRTUALENVS_CREATE=false \
     POETRY_CACHE_DIR='/var/cache/pypoetry' \
     GOOGLE_APPLICATION_CREDENTIALS='/usr/local/airflow/service-account.json'
 
+USER root
 RUN apt-get update \
     && apt-get install -y --no-install-recommends git \
-    # 1. if you don't need postgres, remember to remove postgresql-dev and sqlalchemy
-    # 2. libglib2.0-0 libsm6 libxext6 libxrender-dev libgl1-mesa-dev are required by opencv
-    # 3. git is required by pip install git+https
-    && pip install --no-cache-dir poetry==1.0.5 \
     && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
-
+    && rm -rf /var/lib/apt/lists/* \
+    && chmod -R 777 /usr/local/lib/python3.8/site-packages \
+    && chmod 777 -R /usr/local/include/python3.8/ \
+    && chmod -R 777 /usr/local/bin/
+USER airflow
 COPY pyproject.toml pyproject.toml
 COPY poetry.lock poetry.lock
-RUN python -m poetry install --no-interaction --no-ansi --no-dev \
-    # Cleaning poetry installation's cache for production:
+RUN pip install --no-cache-dir poetry==1.0.5 \
```

Comment:
I remember using pip install to install poetry is not its official recommendation.

Reply:
`curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -` from https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions